src/scales/scale.logarithmic.js

```diff
@@ -179,62 +179,94 @@ module.exports = function(Chart) {
 
+			if (reverse) {
+				range = {
+					start: me.end,
+					end: me.start
+				};
+				sign = -1;
+			} else {
+				range = {
+					start: me.start,
+					end: me.end
+				};
+				sign = 1;
+			}
+			baseValue = range.start;
 			if (me.isHorizontal()) {
-				range = helpers.log10(me.end) - helpers.log10(start); // todo: if start === 0
-				if (newVal === 0) {
-					pixel = me.left;
-				} else {
-					innerDimension = me.width;
-					pixel = me.left + (innerDimension / range * (helpers.log10(newVal) - helpers.log10(start)));
-				}
+				innerDimension = me.width;
+				pixel = (reverse)
+					? me.right
+					: me.left;
 			} else {
-				// Bottom - top since pixels increase downward on a screen
 				innerDimension = me.height;
-				if (start === 0 && !tickOpts.reverse) {
-					range = helpers.log10(me.end) - helpers.log10(me.minNotZero);
-					if (newVal === start) {
-						pixel = me.bottom;
-					} else if (newVal === me.minNotZero) {
-						pixel = me.bottom - innerDimension * 0.02;
-					} else {
-						pixel = me.bottom - innerDimension * 0.02 - (innerDimension * 0.98 / range * (helpers.log10(newVal) - helpers.log10(me.minNotZero)));
-					}
-				} else if (me.end === 0 && tickOpts.reverse) {
-					range = helpers.log10(me.start) - helpers.log10(me.minNotZero);
-					if (newVal === me.end) {
-						pixel = me.top;
-					} else if (newVal === me.minNotZero) {
-						pixel = me.top + innerDimension * 0.02;
-					} else {
-						pixel = me.top + innerDimension * 0.02 + (innerDimension * 0.98 / range * (helpers.log10(newVal) - helpers.log10(me.minNotZero)));
-					}
-				} else if (newVal === 0) {
-					pixel = tickOpts.reverse ? me.top : me.bottom;
-				} else {
-					range = helpers.log10(me.end) - helpers.log10(start);
-					innerDimension = me.height;
-					pixel = me.bottom - (innerDimension / range * (helpers.log10(newVal) - helpers.log10(start)));
+				sign *= -1; // invert, since the upper-left corner of the canvas is at pixel (0, 0)
+				pixel = (reverse)
+					? me.top
+					: me.bottom;
+			}
+			if (realValue !== baseValue) {
+				if (baseValue === 0) { // include zero tick
+					offset = helpers.getValueOrDefault(
+						me.options.ticks.fontSize,
+						Chart.defaults.global.defaultFontSize
+					);
+					innerDimension -= offset;
+					range.start = firstTickValue;
+				}
+				if (realValue !== 0) {
+					offset += innerDimension / (lg(range.end) - lg(range.start)) * (lg(realValue) - lg(range.start));
 				}
+				pixel += sign * offset;
 			}
 			return pixel;
 		},
 		getValueForPixel: function(pixel) {
 			var me = this;
-			var range = helpers.log10(me.end) - helpers.log10(me.start);
-			var value, innerDimension;
+			var reverse = me.options.ticks.reverse;
+			var lg = helpers.log10;
+			var exp = Math.floor(lg(me.minNotZero));
+			var significand = Math.floor(me.minNotZero / Math.pow(10, exp));
+			var firstTickValue = significand * Math.pow(10, exp);
+			var range = (reverse)
```

Comment:
Same here, can we avoid nesting `start` and `end` under `range`?

Reply:
The minifier reduces the code to `r=d?l.right:l.left`, removing the parenthesis.

Being dyslexic, I have trouble reading lines like the following from the function `determineDataLimits`:
`me.min = me.min === null ? minVal : Math.min(me.min, minVal);`
For however short, every time it takes me about 10 seconds to understand what it does.
Expanding it over multiple lines and adding parenthesis removes this readability obstacle.

    me.min = (me.min === null)
        ? minVal
        : Math.min(me.min, minVal);

Still it is just a personal preference.

Reply:
It's really minor and I don't have a strong opinion on which form is better (I usually go for the one which is consistent with the rest of the code if that makes sense).

Anyway, that wasn't my point for this line, but to flatten the useless `range` object:

    var start = reverse ? me.end : me.start;
    var end = reverse ? me.start : me.end;